react/components/OrganizationsAutocomplete.tsx

```diff
@@ -24,59 +24,96 @@ interface Props {
 const OrganizationsAutocomplete = ({ onChange, organizationId }: Props) => {
   const { formatMessage } = useIntl()
   const [term, setTerm] = useState('')
-  const [hasChanged, setHasChanged] = useState(false)
-  const [values, setValues] = useState([] as any)
+  const [debouncedTerm, setDebouncedTerm] = useState(term)
+
+  const [values, setValues] = useState<Array<{ value: string; label: string }>>(
+    []
+  )
+
   const { data, loading, refetch } = useQuery(GET_ORGANIZATIONS, {
     variables: initialState,
     ssr: false,
     notifyOnNetworkStatusChange: true,
   })
 
-  const { data: organization } = useQuery(GET_ORGANIZATION_BY_ID, {
-    variables: { id: organizationId },
-    ssr: false,
-    fetchPolicy: 'network-only',
-    notifyOnNetworkStatusChange: true,
-    skip: !organizationId,
-  })
+  const { data: organization, loading: orgLoading } = useQuery(
+    GET_ORGANIZATION_BY_ID,
+    {
+      variables: { id: organizationId },
+      ssr: false,
+      fetchPolicy: 'network-only',
+      notifyOnNetworkStatusChange: true,
+      skip: !organizationId,
+    }
+  )
 
-  const options = {
-    onSelect: (value: any) => onChange(value),
-    loading,
-    value: values,
-  }
+  useEffect(() => {
+    const handler = setTimeout(() => {
+      setDebouncedTerm(term)
+    }, 500) // 500ms delay
 
-  const onClear = () => {
-    if (!hasChanged) return
-    setTerm('')
-    onChange({ value: null, label: '' })
-  }
+    return () => clearTimeout(handler)
+  }, [term])
 
   useEffect(() => {
-    if (!organization) {
-      return
+    if (debouncedTerm.length > 2) {
+      refetch({
+        ...initialState,
+        search: debouncedTerm,
+      })
+    } else if (debouncedTerm === '') {
+      refetch({
+        ...initialState,
+        search: '',
+      })
     }
+  }, [debouncedTerm, refetch])
 
-    const { name, id } = organization.getOrganizationById
+  useEffect(() => {
+    // eslint-disable-next-line vtex/prefer-early-return
```

Comment:
do we really need to disable lint here? why can't we keep the if/return?

Reply:
lint comment removed @enzomerca